File: data/code_pairs/cpp/1765.cpp
#ifndef MATRIX_OPERATIONS_H
#define MATRIX_OPERATIONS_H

#include <vector>

using Matrix = std::vector<std::vector<double>>;

Matrix matmul(const Matrix& a, const Matrix& b);
Matrix transpose(const Matrix& m);
bool approxEqual(const Matrix& a, const Matrix& b, double eps = 1e-9);
void mass_matrix(Matrix& MM1);

#endif // MATRIX_OPERATIONS_H

File: data/code_pairs/fortran/1765.f90
module matrix_operations_mod
    implicit none
contains
    subroutine mass_matrix(MM1)
        double precision, parameter :: m1 = 1.d0
        double precision, dimension(3,2), parameter :: A1 = reshape([1.d0,0.d0, &
             0.d0,0.d0,1.d0,0.d0],[3,2])
        double precision, dimension(2,2), parameter :: Mel = reshape([1.d0/3.d0, 0.d0, &
             0.d0, 1.d0/3.d0],[2,2])

        double precision, dimension(3,3) :: MM1

        MM1 = m1 * matmul(A1, matmul(Mel, transpose(A1)))
    end subroutine mass_matrix
end module matrix_operations_mod